tests/rgen/Microsoft.Macios.Generator.Tests/Classes/Data/ExpectedMethodsTests.cs

```diff
@@ -242,6 +242,8 @@ public virtual unsafe partial void CompleteRequest (global::Foundation.NSExtensi
 	[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
 	public virtual unsafe partial global::Foundation.NSArray Filter (global::Foundation.NSPredicate predicate)
 	{
+		if (predicate is null)
+			global::ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (predicate));
```

Comment:
This is redundant, `GetNonNullHandle` just below does the same check.

Reply:
here is my train of though, GetNonNullHandle will throw an exception, therefore we could have:
```csharp
if (string is null)
    throw new ArgumentNullException ();
var nspath = global::CoreFoundation.CFString.CreateNative (path);\
var value__handle__ = value!.GetNonNullHandle (nameof (value)); 
```
If the GetNonNullHandle throws, we are leaking the nspath. I think the correct way will be to remove the GetNonNullHandle and move it to a GetHandle and leave the if.

Reply:
I have updated the code to use a priority queue to decide the order of the conversions so that we can avoid issues with the GetNonNullHandle. I don't think the addition will give use a huge penalty since usually methods have very few arguments.